pkgs/development/compilers/llvm/common/flang/default.nix

```diff
@@ -0,0 +1,105 @@
+{
+  lib,
+  stdenv,
+  llvm_meta,
+  monorepoSrc,
+  release_version,
+  runCommand,
+  cmake,
+  libxml2,
+  libllvm,
+  ninja,
+  libffi,
+  libclang,
+  clang,
+  mlir,
+  version,
+  python3,
+  buildLlvmTools,
+  patches ? [ ],
+  devExtraCmakeFlags ? [ ],
+  fetchpatch,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "flang";
+  inherit version;
+
+  src = runCommand "flang-src-${version}" { inherit (monorepoSrc) passthru; } ''
+    mkdir -p "$out"
+    cp -r ${monorepoSrc}/${finalAttrs.pname} "$out"
+    cp -r ${monorepoSrc}/cmake "$out"
+    cp -r ${monorepoSrc}/llvm "$out"
+    cp -r ${monorepoSrc}/clang "$out"
+    cp -r ${monorepoSrc}/mlir "$out"
+    cp -r ${monorepoSrc}/third-party "$out"
+    chmod -R +w $out/llvm
+  '';
+
+  patches =
+    [ ]
```

Comment:
Given the next line gets formatted differently to the following optionals call anyway, not sure there's any need for this empty list.

Reply:
done